javascript/ql/src/Declarations/DeadStoreOfProperty.ql

```diff
@@ -0,0 +1,142 @@
+/**
+ * @name Useless assignment to property
+ * @description An assignment to a property whose value is always overwritten has no effect.
+ * @kind problem
+ * @problem.severity warning
+ * @id js/useless-assignment-to-property
+ * @tags maintainability
+ * @precision high
+ */
+
+import javascript
+import Expressions.DOMProperties
+import DeadStore
+
+/**
+ * Holds if `assign` definitely assigns property `name` of `base`.
```

Comment:
It's not very clear what "definitely" refers to in this comment. I struggle to find a clearer way of putting it, but it needs to be made clear somehow that what we're ensuring here is that the write happens on `base` (as opposed to, say, ensuring that the write refers to `name`, or that the write definitely must happen, or something like that).

Reply:
The docstring is a bit more operational now.